### PR TITLE
Account memory for buffer sizes

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java [hunk 1]
```diff
@@ -58,8 +58,8 @@
 @State(Scope.Thread)
 @OutputTimeUnit(SECONDS)
 @Fork(3)
-@Warmup(iterations = 5, time = 500, timeUnit = TimeUnit.MILLISECONDS)
```
Comment (pnowojski) [inline presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java]: should that be part of this change? or maybe it shouldn't be changed?

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java [hunk 2]
```diff
@@ -193,7 +193,7 @@ public void close()
         hashAggregationBuilder.setOutputPartial();
 
         if (!spiller.isPresent()) {
-            spiller = Optional.of(spillerFactory.create(hashAggregationBuilder.buildTypes(), () -> operatorContext.getSpillContext().newLocalSpillContext()));
+            spiller = Optional.of(spillerFactory.create(hashAggregationBuilder.buildTypes(), () -> operatorContext.getSpillContext().newLocalSpillContext(), operatorContext.getSystemMemoryContext().newAggregatedMemoryContext()));
```
Comment (pnowojski) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java:196]: reformat this line

File: presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpiller.java [hunk 3]
```diff
@@ -124,8 +134,10 @@ private void writePages(PagesSerde.PagesWriter pagesWriter, Iterator<Page> pages
     {
         try {
             InputStream input = new FileInputStream(targetFileName.toFile());
+            memoryContext.setBytes(BUFFER_SIZE);
```
Comment (pnowojski) [inline presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpiller.java:137]: Underlying method of `setBytes` - `com.facebook.presto.operator.DriverContext#reserveSystemMemory` is not thread safe and you are accessing it from multiple threads.

**Review by pnowojski — CHANGES_REQUESTED** [review]

File: presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpiller.java [hunk 4]
```diff
@@ -123,9 +133,11 @@ private void writePages(PagesSerde.PagesWriter pagesWriter, Iterator<Page> pages
     private Iterator<Page> readPages()
     {
         try {
-            InputStream input = new BufferedInputStream(new FileInputStream(targetFileName.toFile()));
+            InputStream input = new FileInputStream(targetFileName.toFile());
+            memoryContext.setBytes(BUFFER_SIZE);
```
Comment (pnowojski) [inline presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpiller.java:137]: Underlying method of `setBytes` - `com.facebook.presto.operator.DriverContext#reserveSystemMemory` is not thread safe and you are accessing it from multiple threads.

File: presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java (hunk 1 above)
Comment (losipiuk) [reply presto-main/src/test/java/com/facebook/presto/operator/spiller/BenchmarkBinaryFileSpiller.java]: done

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java (hunk 2 above)
Comment (losipiuk) [reply presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java:196]: done

File: presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpiller.java (hunk 4 above)
Comment (losipiuk) [reply presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpiller.java:137]: discussed

**Comment by losipiuk** [comment]
merged

**Pull request closed without merging**